core/unit_test/TestViewAPI_e.hpp

```diff
@@ -240,6 +240,32 @@ struct TestViewOverloadResolution {
 TEST(TEST_CATEGORY, view_overload_resolution) {
   TestViewOverloadResolution<TEST_EXECSPACE>::test_function_overload();
 }
+
+template <typename MemorySpace>
+struct TestViewAllocationLargeRank {
+  using ViewType = Kokkos::View<char********, MemorySpace>;
+
+  KOKKOS_FUNCTION void operator()(int) const {
+    size_t idx = v.extent(0) - 1;
+    auto& lhs  = v(idx, idx, idx, idx, idx, idx, idx, idx);
+    lhs        = 0;  // This is where it segfaulted
```

Comment:
Could this assign a non-zero value `foo`, and then assert on the host that `v.access(idx, ...) == foo`? As written, this is a weak smoke-test that the particular case doesn't happen to crash in our CI environments